sklearn/tree/tests/test_tree.py

```diff
@@ -2406,3 +2407,25 @@ def test_splitter_serializable(Splitter):
     splitter_back = pickle.loads(splitter_serialize)
     assert splitter_back.max_features == max_features
     assert isinstance(splitter_back, Splitter)
+
+
+def test_tree_deserialization_from_read_only_buffer():
+    # Non-regression test for the issue:
+    # https://github.com/scikit-learn/scikit-learn/issues/25584
+    # This test helps to ensure that the Tree module is able to
+    # successfully deserialize data from read only buffers.
+    X, y = fetch_california_housing(return_X_y=True, as_frame=True)
+    features = ["MedInc", "AveOccup", "HouseAge", "AveRooms"]
+    est = DecisionTreeRegressor()
+    est.fit(X, y)
+
+    PartialDependenceDisplay.from_estimator(
+        est,
+        X,
+        features,
+        kind="individual",
+        subsample=50,
+        n_jobs=3,
+        grid_resolution=20,
+        random_state=0,
+    )
```

Comment:
Instead of indirectly relying on joblib parallelism via partial dependence computation, it would be better to write a non-regression test based on unpickling from a readonly buffer. This can be done as explained in this comment https://github.com/scikit-learn/scikit-learn/issues/25584#issuecomment-1425795214.

To do this in a test, you could use the `tmpdir` fixture from pytest.

Also this would be great to write this test as a new estimator check to be run by `test_common.py` for any scikit-learn estimator, not just decision trees.

I thought we already had such a common test but apparently this is not the case.

Reply:
Indeed.

I need to improve my knowledge of common tests; ideally we must assert Cython implementations' correct behavior on (memmapped) readonly datasets.

@OmarManzoor: do you see how you can perform what Olivier has mentioned?

Reply:
I did check the common tests and from what I understand we need to replicate the behavior in this PR to all estimators in the common tests.

Reply:
Yes, for the scope of this PR, you can stick to the remarks in the two first paragraph of https://github.com/scikit-learn/scikit-learn/pull/25585#discussion_r1102757421.

Then, the general coverage explained in the third and fourth paragraphs of https://github.com/scikit-learn/scikit-learn/pull/25585#discussion_r1102757421 better be treated in another pull request IMO. I think you can author another PR for it if you are interested but you need not to for this current PR to be accepted.

Reply:
Okay sounds good. I think this PR can be merged then?

Reply:
I let @ogrisel merge when this LGTH (I am not 100% sure that the current state of the test satisfies what he indicated in this thread first comment).